packages/wallet-manager/src/WalletManagerController.ts

```diff
@@ -0,0 +1,162 @@
+import { ReactiveController, ReactiveControllerHost } from 'lit';
+import { EvmWallet, SubstrateWallet } from '.';
+import { Web3Provider } from '@ethersproject/providers';
+import { ApiPromise } from '@polkadot/api';
+import { AddChain } from './types';
+import { IWalletManagerController } from './interfaces';
+import { Signer } from '@ethersproject/abstract-signer';
+
+export class WalletManagerController implements IWalletManagerController {
+  host: ReactiveControllerHost;
+  evmWallet?: EvmWallet;
+  substrateWallet?: SubstrateWallet;
+  account?: string;
+  substrateAccount?: string;
+
+  constructor(host: ReactiveControllerHost) {
+    (this.host = host).addController(this as ReactiveController);
+  }
+
+  private appendProviderEvents(evmWallet: EvmWallet): void {
+    evmWallet.addListener('walletAccountChanged', (account) => {
+      this.account = account;
+      this.host.requestUpdate();
+    });
+
+    evmWallet.addListener('walletChainChanged', () => {
+      this.account = evmWallet.address;
+      this.host.requestUpdate();
+    });
+  }
+
+  public addWalletAccountChangeEventListener(
+    callback: (account: string) => void
+  ): void {
+    this.evmWallet?.addListener('walletAccountChanged', (account) => {
+      this.account = account;
+      callback(account);
+    });
+  }
+
+  public addWalletChangedEventListener(callback: () => void): void {
+    this.evmWallet?.addListener('walletChainChanged', () => {
+      this.account = this.evmWallet?.address;
+      callback();
+    });
+  }
+
+  /**
+   * @name initFromWeb3Provider
+   * @param web3Provider Web3Provider
+   * @description Initializes the EvmWallet from a Web3Provider
+   */
+  public initFromWeb3Provider(web3Provider: Web3Provider): void {
+    this.evmWallet = EvmWallet.initFromWeb3Provider(web3Provider);
```

Comment:
I think is more clear having the static method considering the branch condition on the constructor for the `emv` class

Reply:
how exactly is it more clear? you're instantiating new wallet, isn't it more clear with the `new` keyword? this looks like you're initialising (!= creating) some wallet.

also, having two separate static methods that call constructor with and without given parameter seems a bit absurd, since all this can be done with just the constructor.

Reply:
done